lib/msal-common/src/client/AuthorizationCodeClient.ts

```diff
@@ -0,0 +1,163 @@
+     */
+    async acquireToken(request: AuthorizationCodeRequest): Promise<string> {
+
+        const authority: Authority = await this.createAuthority(request && request.authority);
+        const response = await this.executeTokenRequest(authority, request);
+        return JSON.stringify(response.body);
+        // TODO add response_handler here to send the response
+    }
+
+    /**
+     * Executes POST request to token endpoint
+     * @param authority
+     * @param request
+     */
+    private async executeTokenRequest(authority: Authority, request: AuthorizationCodeRequest): Promise<NetworkResponse<ServerAuthorizationTokenResponse>> {
+
+        const requestBody = this.createTokenRequestBody(request);
+        const headers: Map<string, string> = this.createDefaultTokenRequestHeaders();
+
+        return this.executePostToTokenEndpoint(authority.tokenEndpoint, requestBody, headers);
+    }
+
+    /**
+     * Generates a map for all the params to be sent to the service
+     * @param request
+     */
+    private createTokenRequestBody(request: AuthorizationCodeRequest) : string {
+        const parameterBuilder = new RequestParameterBuilder();
+
+        parameterBuilder.addClientId(this.config.authOptions.clientId);
+
+        // validate the redirectUri (to be a non null value)
+        RequestValidator.validateRedirectUri(request.redirectUri);
+        parameterBuilder.addRedirectUri(request.redirectUri);
+
+        const scopeSet = new ScopeSet(
+            request.scopes || [],
+            this.config.authOptions.clientId,
+            false);
+        parameterBuilder.addScopes(scopeSet);
+
+        // add code: user set, not validated
+        parameterBuilder.addAuthorizationCode(request.code);
+
+        // add code_verifier if passed
+        if (request.codeVerifier) {
+            parameterBuilder.addCodeVerifier(request.codeVerifier);
+        }
+
+        parameterBuilder.addGrantType(GrantType.AUTHORIZATION_CODE_GRANT);
+
+        return parameterBuilder.createQueryString();
+    }
+
+    /**
+     * This API validates the `AuthorizationCodeUrlRequest` and creates a URL
+     * @param request
+     */
+    private createAuthCodeUrlQueryString(request: AuthorizationCodeUrlRequest): string {
+        const parameterBuilder = new RequestParameterBuilder();
+
+        parameterBuilder.addClientId(this.config.authOptions.clientId);
+
+        const scopeSet = new ScopeSet(request.scopes || [],
+            this.config.authOptions.clientId,
+            false);
+        parameterBuilder.addScopes(scopeSet);
+
+        // validate the redirectUri (to be a non null value)
+        RequestValidator.validateRedirectUri(request.redirectUri);
+        parameterBuilder.addRedirectUri(request.redirectUri);
+
+        // generate the correlationId if not set by the user and add
+        const correlationId = request.correlationId
+            ? request.correlationId
+            : this.config.cryptoInterface.createNewGuid();
```

Comment:
`const correlationId = request.correlationId ||  this.config.cryptoInterface.createNewGuid()`

Reply:
updated